### PR TITLE
Makes it possible to attach igniter assemblies to the heart + Minor QOL

### PR DESCRIPTION
**Pull request by nemvar**
## About The Pull Request

Makes it possible to attach igniter assemblies to the heart, making it stop/resume beating when signaled.

Also makes the cursed heart give you some more leeway before you need to pump it for the first time.

## Why It's Good For The Game

Adds a little bit of FUN to the game. Makes it possible to torture, restrain, blackmail and cheese heart attacks in new and interesting ways. Since actually getting a signaler onto a heart is quite a bit of work, I don't think this has any big balance implication.

Note that a lot of assemblies don't actually do anything when implanted. Only signalers and timers work.

## Changelog
:cl:
add: You can now attach igniter assemblies to hearts, resulting in them stopping/resuming beating when signaled.
tweak: The cursed heart is a little more forgiving when for inserted. You won't die as hard as before if you missed the first few ticks due to lag.
/:cl:

### CODE REVIEW

**Comment by Barhandar** [comment]
Getting shocked with assembled heart should explode it.

**Comment by anconfuzedrock** [comment]
I hope this has use in restarting hearts that are stopped through other means.

**Comment by OverlyGeneric** [comment]
> I hope this has use in restarting hearts that are stopped through other means.

I suppose it would act like an artificial cardiac pacemaker which regulates the heartbeat.

A strong electric shock could even disable the pacemaker which leads to cardiac arrest while a mild one could cause muscles in the heart to rupture.

**Comment by nemvar** [comment]
You can restart hearts that have been stopped by hiveminds with this. It does basically nothing against actual cardiac arrest disease from the random event.

**Comment by necromanceranne** [comment]
Health sensor ignitor assemblies would be basically a pacemaker.

**Comment by Barhandar** [comment]
@nemvar it should work at least like actual shocks and defibs for the disease, and yes, health sensor should work for this and trigger on said cardiac arrests and hivemind stoppers.

**Comment by nemvar** [comment]
Apparently this does fix the cardiac arrest disease. I can't really get health sensors to work unless I were to rework the way internal organs are handled. You will have to carry a second assembly with a signaler in your bag for now.

**Comment by Goodfishie** [comment]
Just a note, cardiac arrest =/= heart attack

Heart attack - Blockage in heart
Cardiac Arrest - Heart not beating/beating properly

Heart attacks can cause cardiac arrest but are not arrests on their own

I'm pretty sure the disease is called Heart Disease or Heart Attack or something

**Comment by necromanceranne** [comment]
Actually...here's an important question. How many times could a strangulation induced defibbing could you come back from if you had a igniter assembly attached with a health sensor set to 'death'.

**Comment by Razharas** [comment]
Assembly handling really should be componentised to be quite honest with you famiglia

File: code/modules/surgery/organs/heart.dm [hunk 1]
```diff
@@ -9,6 +9,47 @@
...
+	return TRUE
+	
+/obj/item/organ/heart/receive_signal()
+	if(beating)
+		Stop()
+	else
+		Restart()
+		addtimer(CALLBACK(src, .proc/stop_if_unowned), 80) //The heart is a vunerable muscle.
+
+/obj/item/organ/heart/Crossed(atom/movable/AM)
```
Comment (optimumtact) [inline code/modules/surgery/organs/heart.dm]: there's a signal, register for it

bruh

File: code/modules/surgery/organs/heart.dm [hunk 2]
```diff
@@ -27,6 +68,9 @@
 
 /obj/item/organ/heart/attack_self(mob/user)
 	..()
+	if(assembly)
+		assembly.attack_self(user)
```
Comment (optimumtact) [inline code/modules/surgery/organs/heart.dm]: pretty sure there's a signal for this too

Comment (ShizCalev) [reply code/modules/surgery/organs/heart.dm]: COMSIG_ITEM_ATTACK_SELF

File: code/modules/surgery/organs/heart.dm (hunk 1 above)
Comment (ShizCalev) [reply code/modules/surgery/organs/heart.dm]: COMSIG_MOVABLE_CROSSED

**Pull request closed without merging**